src/Core/Domain/Discount/Command/UpdateDiscountCommand.php

```diff
@@ -133,6 +140,8 @@ public function setValidityDateRange(DateTimeImmutable $from, DateTimeImmutable
 
         $this->validFrom = $from;
         $this->validTo = $to;
+        $this->markDirty('validFrom');
+        $this->markDirty('validTo');
```

Comment:
I guess this will be useful for `validTo` that should also be turned into a nullable in another PR